packages/rum/src/performanceCollection.ts

```diff
@@ -130,18 +158,23 @@ function computeRelativePerformanceTiming() {
 }
 
 function handlePerformanceEntries(lifeCycle: LifeCycle, entries: PerformanceEntry[]) {
-  function notify(entry: PerformanceEntry) {
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED, entry)
-  }
-
-  entries.filter((entry) => entry.entryType === 'resource').forEach(notify)
+  entries.forEach((entry) => {
+    if (
+      entry.entryType === 'resource' ||
+      entry.entryType === 'navigation' ||
+      entry.entryType === 'paint' ||
+      entry.entryType === 'longtask'
+    ) {
+      handleRumPerformanceEntry(lifeCycle, entry as RumPerformanceEntry)
+    }
+  })
+}
 
-  entries
-    .filter((entry) => entry.entryType === 'navigation')
-    // Exclude incomplete navigation entries by filtering out those who have a loadEventEnd at 0
-    .filter((entry) => (entry as PerformanceNavigationTiming).loadEventEnd > 0)
-    .forEach(notify)
+function handleRumPerformanceEntry(lifeCycle: LifeCycle, entry: RumPerformanceEntry) {
```

Comment:
```suggestion
function handlePerformanceEntry(lifeCycle: LifeCycle, entry: RumPerformanceEntry) {
```
or rename parent to `handleRumPerformanceEntries` for consistency

Reply:
I'm not sure about this. The parent is accepting `PerformanceEntry[]`, and this function is accepting `RumPerformanceEntry`. The naming reflects this difference.

Reply:
OK make sense